cmd/libsnap-confine-private/utils.c

```diff
@@ -217,3 +217,15 @@ int sc_nonfatal_mkpath(const char *const path, mode_t mode)
 	}
 	return 0;
 }
+
+int sc_must_parse_int(const char *s)
+{
+	char *endptr = NULL;
+
+	errno = 0;
+	long i = strtol(s, &endptr, 10);
+	if (errno != 0 || s == endptr || *endptr != '\0' || i < 0) {
+		die("cannot parse number in '%s'", s);
+	}
```

Comment:
This function should be renamed sc_must_parse_positive_int() if you are going to error on 'i < 0'.

Reply:
Thanks, indeed! I will just make it parse any int and then the caller can check for< 0 if needed.